crates/proof-of-sql/src/base/database/column_operation.rs

```diff
@@ -0,0 +1,202 @@
+                Ok(Self::Int128(add_vecs_with_casting(lhs, rhs)))
+            }
+            (Self::SmallInt(lhs), Self::Decimal75(precision, scale, rhs)) => {
+                add_decimal_columns(lhs, rhs, 5, 0, precision.value(), scale)
+            }
+            (Self::SmallInt(lhs), Self::Scalar(rhs)) => {
+                Ok(Self::Scalar(add_vecs_with_casting(lhs, rhs)))
+            }
+            (Self::Int(lhs), Self::SmallInt(rhs)) => Ok(Self::Int(add_vecs_with_casting(rhs, lhs))),
+            (Self::Int(lhs), Self::Int(rhs)) => Ok(Self::Int(add_vecs(lhs, rhs))),
+            (Self::Int(lhs), Self::BigInt(rhs)) => {
+                Ok(Self::BigInt(add_vecs_with_casting(lhs, rhs)))
+            }
+            (Self::Int(lhs), Self::Int128(rhs)) => {
+                Ok(Self::Int128(add_vecs_with_casting(lhs, rhs)))
+            }
+            (Self::Int(lhs), Self::Decimal75(precision, scale, rhs)) => {
+                add_decimal_columns(lhs, rhs, 9, 0, precision.value(), scale)
+            }
+            (Self::Int(lhs), Self::Scalar(rhs)) => {
+                Ok(Self::Scalar(add_vecs_with_casting(lhs, rhs)))
+            }
+            (Self::BigInt(lhs), Self::SmallInt(rhs)) => {
+                Ok(Self::BigInt(add_vecs_with_casting(rhs, lhs)))
+            }
+            (Self::BigInt(lhs), Self::Int(rhs)) => {
+                Ok(Self::BigInt(add_vecs_with_casting(rhs, lhs)))
+            }
+            (Self::BigInt(lhs), Self::BigInt(rhs)) => Ok(Self::BigInt(add_vecs(lhs, rhs))),
+            (Self::BigInt(lhs), Self::Int128(rhs)) => {
+                Ok(Self::Int128(add_vecs_with_casting(lhs, rhs)))
+            }
+            (Self::BigInt(lhs), Self::Decimal75(precision, scale, rhs)) => {
+                add_decimal_columns(lhs, rhs, 19, 0, precision.value(), scale)
+            }
+            (Self::BigInt(lhs), Self::Scalar(rhs)) => {
+                Ok(Self::Scalar(add_vecs_with_casting(lhs, rhs)))
+            }
+            (Self::Int128(lhs), Self::SmallInt(rhs)) => {
+                Ok(Self::Int128(add_vecs_with_casting(rhs, lhs)))
+            }
+            (Self::Int128(lhs), Self::Int(rhs)) => {
+                Ok(Self::Int128(add_vecs_with_casting(rhs, lhs)))
+            }
+            (Self::Int128(lhs), Self::BigInt(rhs)) => {
+                Ok(Self::Int128(add_vecs_with_casting(rhs, lhs)))
+            }
+            (Self::Int128(lhs), Self::Int128(rhs)) => Ok(Self::Int128(add_vecs(lhs, rhs))),
+            (Self::Int128(lhs), Self::Decimal75(precision, scale, rhs)) => {
+                add_decimal_columns(lhs, rhs, 39, 0, precision.value(), scale)
+            }
+            (Self::Int128(lhs), Self::Scalar(rhs)) => {
+                Ok(Self::Scalar(add_vecs_with_casting(lhs, rhs)))
+            }
+            (Self::Decimal75(precision, scale, lhs), Self::SmallInt(rhs)) => {
+                add_decimal_columns(rhs, lhs, 5, 0, precision.value(), scale)
```

Comment:
This is what I meant. I didn't look close enough on the previous comment.
```suggestion
                add_decimal_columns(rhs, lhs, precision.value(), scale, 5, 0)
```

Reply:
Oh, never mind, I see that you have `rhs` coming first here.